qcodes/instrument_drivers/Keysight/KtMAwg.py

```diff
@@ -0,0 +1,319 @@
+            raise ValueError(f"Invalid channel: {chan}, expecting ch1:ch3")
+
+        super().__init__(parent, name)
+        self._channel = ctypes.create_string_buffer(
+            "Channel{}".format(chan).encode("ascii"))
+
+        # Used to access waveforms loaded into the driver
+        self._awg_handle = None
+
+        # FIXME is this ok?
+        self._parent = parent
+        self._catch_error = self._parent._catch_error
+
+        self.add_parameter('output_term_config',
+                           label="Output Terminal Configuration",
+                           get_cmd=partial(parent.get_vi_int,
+                                           KTMAWG_ATTR_TERMINAL_CONFIGURATION,
+                                           ch=self._channel),
+                           set_cmd=partial(parent.set_vi_int,
+                                           KTMAWG_ATTR_TERMINAL_CONFIGURATION,
+                                           ch=self._channel),
+                           val_mapping={'differential': KTMAWG_VAL_TERMINAL_CONFIGURATION_DIFFERENTIAL,
+                                        'single': KTMAWG_VAL_TERMINAL_CONFIGURATION_SINGLE_ENDED})
+
+        self.add_parameter('operation',
+                           label="Operating Mode",
+                           get_cmd=partial(parent.get_vi_int,
+                                           KTMAWG_ATTR_OPERATION_MODE,
+                                           ch=self._channel),
+                           set_cmd=partial(parent.set_vi_int,
+                                           KTMAWG_ATTR_OPERATION_MODE,
+                                           ch=self._channel),
+                           val_mapping={'continuous': KTMAWG_VAL_OPERATE_CONTINUOUS,
+                                        'burst': KTMAWG_VAL_OPERATE_BURST}
+                           )
+
+        self.add_parameter('output',
+                           label="Output Enable",
+                           get_cmd=partial(parent.get_vi_bool,
+                                           KTMAWG_ATTR_OUTPUT_ENABLED,
+                                           ch=self._channel),
+                           set_cmd=partial(parent.set_vi_bool,
+                                           KTMAWG_ATTR_OUTPUT_ENABLED,
+                                           ch=self._channel),
+                           val_mapping={"on": 1, "off": 0})
+
+        self.add_parameter('gain_config',
+                           label="AWG Gain Control Mode",
+                           set_cmd=self._set_gain_control,
+                           get_cmd=self._get_gain_control,
+                           val_mapping={"composite": KTMAWG_VAL_GAIN_CONTROL_COMPOSITE,
+                                        "component": KTMAWG_VAL_GAIN_CONTROL_COMPONENT})
+        self.add_parameter('gain',
+                           label="Composite Output Gain",
+                           set_cmd=self._set_gain,
+                           get_cmd=None)
```

Comment:
Validator is missing. You can add possible values for example as `vals=Numbers(min, max)`